### PR TITLE
Closes #19. Add authenticator integration.

### LINKED ISSUE #19: Add authenticator integration.

### PR DESCRIPTION
**Pull request by artemii235**
Things that were done:
1. Integrated authenticator.
2. Split verification services to different files to improve code readability.
3. Changed package manager to yarn.
4. Fixed tslint.
5. InMemory storage stores JSON serialized objects now - it stored pointer to object that caused some hard to investigate issues when you change the object in another place.
6. Storage `set` method now might store object permanently depending on input parameters.
7. Initiate endpoint validation moved to middleware - we need different rules depending on request params and imho it looks better when services themselves don't contain a lot of validation logic.
8. Many of types/interfaces were moved to index.d.ts
9. Add separate tsconfig for production build.

### CODE REVIEW

**Review by hlogeon — COMMENTED** [review]
Давай наверн Exceptions все таки объединим?

**Comment by artemii235** [comment]
@hlogeon Не вопрос, сделаю)

File: src/services/storages.ts [hunk 1]
```diff
@@ -69,20 +31,26 @@ export class SimpleInMemoryStorageService implements StorageService {
   /**
    * @inheritdoc
    */
-  public set<T>(name: string, value: T, options: ValueOptions): Promise<T> {
+  public set<T>(name: string, value: T, options?: ValueOptions): Promise<T> {
```
Comment (AlekNS) [inline src/services/storages.ts:34]: Why need values in storage without TTL?

Comment (artemii235) [reply src/services/storages.ts:34]: @AlekNS We need to store consumer's Authenticator private key permanently until consumer decides to disable 2FA. So I had to refactor `set` method.

File: package.json [hunk 2]
```diff
@@ -33,7 +33,8 @@
     "redis": "2.6.3",
     "reflect-metadata": "0.1.10",
     "request": "2.81.0",
-    "rolling-rate-limiter": "0.1.9"
+    "rolling-rate-limiter": "0.1.9",
+    "typemoq": "^2.0.1"
```
Comment (AlekNS) [inline package.json:37]: May be replace sinon mocks by typemoq for uniformity?

File: src/services/storages.ts (hunk 1 above)
Comment (AlekNS) [reply src/services/storages.ts:34]: Hm... is it of "verification" responsibility to store this persistent data?

File: package.json (hunk 2 above)
Comment (artemii235) [reply package.json:37]: I'm not sure what I like more yet :) TypeMoq was specially designed for Typescript but produces too much code, however I got stuck with sinon - wasn't able to setup mocks due to failing type checks, I will try sinon 1 more time later.

File: src/services/storages.ts (hunk 1 above)
Comment (artemii235) [reply src/services/storages.ts:34]: @AlekNS This secret is required to `verify` 2FA code which is definitely `backend-verification` responsibility. What service should store it if not `verification`?

File: package.json (hunk 2 above)
Comment (hlogeon) [reply package.json:37]: Just move it to it's own issue :)

File: src/services/storages.ts (hunk 1 above)
Comment (hlogeon) [reply src/services/storages.ts:34]: In my opinion, it's verification service's responsibility so it's OK

**Review by hlogeon — APPROVED** [review]

**Review by AlekNS — APPROVED** [review]

**Pull request merged**